src/main/java/org/sagebionetworks/bridge/services/MasterSchedulerService.java

```diff
@@ -0,0 +1,71 @@
+package org.sagebionetworks.bridge.services;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import java.util.List;
+
+import org.sagebionetworks.bridge.dao.MasterSchedulerConfigDao;
+import org.sagebionetworks.bridge.dao.MasterSchedulerStatusDao;
+import org.sagebionetworks.bridge.exceptions.BadRequestException;
+import org.sagebionetworks.bridge.models.schedules.MasterSchedulerConfig;
+import org.sagebionetworks.bridge.validators.MasterSchedulerConfigValidator;
+import org.sagebionetworks.bridge.validators.Validate;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+@Component
+public class MasterSchedulerService {
+    
+    private MasterSchedulerConfigDao schedulerConfigDao;
+
+    @SuppressWarnings("unused")
+    private MasterSchedulerStatusDao schedulerStatusDao;
+
+    @Autowired
+    public final void setSchedulerConfigDao(MasterSchedulerConfigDao schedulerConfigDao) {
+        this.schedulerConfigDao = schedulerConfigDao;
+    }
+    
+    @Autowired
+    public final void setSchedulerStatusDao(MasterSchedulerStatusDao schedulerStatusDao) {
+        this.schedulerStatusDao = schedulerStatusDao;
+    }
+    
+    public List<MasterSchedulerConfig> getAllSchedulerConfigs() {
+        return schedulerConfigDao.getAllSchedulerConfig();
+    }
+
+    public MasterSchedulerConfig createSchedulerConfig(MasterSchedulerConfig schedulerConfig) {
+        checkNotNull(schedulerConfig.getScheduleId());
+        
+        Validate.entityThrowingException(MasterSchedulerConfigValidator.INSTANCE, schedulerConfig);
+        return schedulerConfigDao.createSchedulerConfig(schedulerConfig);
+    }
+
+    public MasterSchedulerConfig getSchedulerConfig(String scheduleId) {
+        checkNotNull(scheduleId);
+        
+        MasterSchedulerConfig schedulerConfig = schedulerConfigDao.getSchedulerConfig(scheduleId);
+        if (schedulerConfig == null) {
+            throw new BadRequestException("Can't get scheduler config for scheduleId=" + scheduleId 
+                    + ": scheduler does not exists");
```

Comment:
I noticed in that in other DAOs the delete() method has the null check and doesn't throw any exception if it is null. Should I handle it similarly in the scheduler?

Reply:
I think in this case, it should throw an error, since this is used as an admin API, and having some feedback when something unexpected happens would be useful.